plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -185,12 +183,26 @@ public CompletionStage<UpdateDataObjectResponse> updateDataObjectAsync(UpdateDat
                 String source = Strings.toString(MediaTypeRegistry.JSON, request.dataObject());
                 JsonNode jsonNode = OBJECT_MAPPER.readTree(source);
                 Map<String, AttributeValue> updateItem = JsonTransformer.convertJsonObjectToDDBAttributeMap(jsonNode);
-                updateItem.put(HASH_KEY, AttributeValue.builder().s(tenantId).build());
-                updateItem.put(RANGE_KEY, AttributeValue.builder().s(request.id()).build());
+                updateItem.remove(HASH_KEY);
+                updateItem.remove(RANGE_KEY);
+                Map<String, AttributeValueUpdate> updateAttributeValue = updateItem
+                    .entrySet()
+                    .stream()
+                    .collect(
+                        Collectors
+                            .toMap(
+                                Map.Entry::getKey,
+                                entry -> AttributeValueUpdate.builder().action(AttributeAction.PUT).value(entry.getValue()).build()
+                            )
+                    );
+                Map<String, AttributeValue> updateKey = new HashMap<>();
+                updateKey.put(HASH_KEY, AttributeValue.builder().s(tenantId).build());
+                updateKey.put(RANGE_KEY, AttributeValue.builder().s(request.id()).build());
                 UpdateItemRequest updateItemRequest = UpdateItemRequest
                     .builder()
                     .tableName(getTableName(request.index()))
-                    .key(updateItem)
+                    .key(updateKey)
```

Comment:
This will conflict with changes in https://github.com/opensearch-project/ml-commons/pull/2605

Reply:
Done. Merged with https://github.com/opensearch-project/ml-commons/pull/2605